packages/snjs/lib/services/item_manager.ts

```diff
@@ -721,6 +718,29 @@ export class ItemManager extends PureService {
     return searchArray(this.tags, { title: title });
   }
 
+  /**
+   * Finds tags with titles starting with a search query
+   */
+  public searchTags(searchQuery: string): SNTag[] {
+    const delimiter = '.';
+    return this.tags
+      .filter((tag) =>
+        tag.title
+          .toLowerCase()
+          .split(delimiter)
+          .some((component) => component.startsWith(searchQuery.toLowerCase()))
+      )
+      .sort(
+        typeof Intl !== 'undefined'
+          ? (a, b) =>
+              new Intl.Collator('en', { numeric: true }).compare(
+                a.title,
+                b.title
+              )
+          : (a, b) => (a.title > b.title ? 1 : -1)
+      );
+  }
```

Comment:
@mobitar would this be the change you're suggesting for reusing collator?

```suggestion
  public searchTags(searchQuery: string): SNTag[] {
    const delimiter = '.';
    const collator = new Intl.Collator('en', { numeric: true });
    return this.tags
      .filter((tag) =>
        tag.title
          .toLowerCase()
          .split(delimiter)
          .some((component) => component.startsWith(searchQuery.toLowerCase()))
      )
      .sort(
        typeof Intl !== 'undefined'
          ? (a, b) =>
              collator.compare(
                a.title,
                b.title
              )
          : (a, b) => (a.title > b.title ? 1 : -1)
      );
  }
```

Reply:
Since it's a stateless object you could potentially move it even higher up, as a constant

Reply:
You likely could use `a.title.localeCompare(b.title)` instead of `a.title > b.title` 🙂